### PR TITLE
Debug when folder deletion fails

### PR DESCRIPTION
**Pull request by mosteo**
Extra output when a file/folder deletion fails, to try and debug occurrences like #1417.

### CODE REVIEW

File: src/alire/alire-directories.adb [hunk 1]
```diff
@@ -253,6 +253,28 @@ package body Alire.Directories is
...
+      when E : others =>
+         Trace.Error ("Could not delete: " & Path);
+         Log_Exception (E, Error);
+         Trace.Error ("Contents follow: ");
+         declare
+            use AAA.Strings;
+            Output : Vector;
+            Code : constant Integer :=
+                     OS_Lib.Subprocess.Unchecked_Spawn_And_Capture
+                       ("ls", To_Vector ("-alRF"),
```
Comment (Fabien-Chouteau) [inline src/alire/alire-directories.adb]: I don't think it's good to rely on and external `ls` command here. On Windows it might not be available.

**Review by Fabien-Chouteau — CHANGES_REQUESTED** [review]

**Comment by mosteo** [comment]
I've added an alternative for Windows. Nonetheless, this is kind of a brute attempt to diagnose exactly the failures happening here... Other ideas welcome :S

**Comment by mosteo** [comment]
I know what the root cause is now and can reproduce locally, working on the fix.

**Comment by mosteo** [comment]
The CI error in this one will go away after #1427

**Review by Fabien-Chouteau — APPROVED** [review]

**Pull request merged**